### PR TITLE
Add section about migrating from `next/image` to `next/future/image`

### PR DESCRIPTION
**Pull request by styfle**
This expands on the previous PR #38978 with a section on migrating

### CODE REVIEW

File: docs/api-reference/next/future/image.md [hunk 1]
```diff
@@ -39,13 +41,64 @@ Compared to `next/image`, the new `next/future/image` component has the followin
...
 
-The default layout for `next/image` was `intrinsic`, which would shrink the `width` if the image was larger than it's container. Since no styles are automatically applied to `next/future/image`, you'll need to add the following CSS to achieve the same behavior:
+## Migration
+
+Although `layout` is not available, you can migrate `next/image` to `next/future/image` using a few props. The following is a comparison of the two components:
 
-```css
-max-width: 100%;
-height: auto;
+**intrinsic: next/image**
```
Comment (ijjk) [inline docs/api-reference/next/future/image.md:48]: Should we use a table to organize the legacy/future comparison a bit more together similar to our page config allowed/not allowed table? https://github.com/vercel/next.js/blob/canary/errors/invalid-page-config.md

Comment (styfle) [reply docs/api-reference/next/future/image.md:48]: Tables don't render nicely right now. For example https://nextjs.org/docs/api-reference/next/image#layout

Comment (styfle) [reply docs/api-reference/next/future/image.md]: @ijjk Ok I think I got a decent table in [b69feb1](https://github.com/vercel/next.js/pull/39270/commits/b69feb136ec23f0c5efd9691c47a57a199411112)

**Review by ijjk — APPROVED** [review]
Thanks!

File: docs/api-reference/next/future/image.md [hunk 2]
```diff
@@ -39,13 +41,134 @@ Compared to `next/image`, the new `next/future/image` component has the followin
...
+    <th>next/future/image</th>
+  </tr>
+</thead>
+<tbody>
+
+<tr>
+<td>
+
+```jsx
+import Image from 'next/image'
```
Comment (leerob) [inline docs/api-reference/next/future/image.md:61]: This looks a bit strange when live.
CleanShot 2022-08-02 at 19 45 07@2x

Thoughts on just doing section headers to for each and just including the after code snippet with a comment in there? 

```jsx
// Replaces default usage of `next/image`
import Image from 'next/future/image'
import img from '../img.png'

const css = { maxWidth: '100%', height: 'auto' }
export default function Page() {
  return <Image src={img} style={css} />
}
```

Comment (leerob) [reply docs/api-reference/next/future/image.md:61]: CleanShot 2022-08-03 at 10 54 38@2x

Comment (ijjk) [reply docs/api-reference/next/future/image.md:61]: > Thoughts on just doing section headers to for each and just including the after code snippet with a comment in there?

Sounds good to me, would be curious if tables would render better if we hid the sidebar 🤔

Comment (styfle) [reply docs/api-reference/next/future/image.md:61]: @leerob @ijjk PR with a quick hack: #39309

**Pull request merged**